src/material/paginator/paginator.ts

```diff
@@ -155,14 +158,20 @@ export abstract class _MatPaginatorBase<
 
   /** The set of provided page size options to display to the user. */
   @Input()
-  get pageSizeOptions(): number[] {
+  get pageSizeOptions(): PageSizeOptions {
     return this._pageSizeOptions;
   }
-  set pageSizeOptions(value: number[] | readonly number[]) {
-    this._pageSizeOptions = (value || []).map(p => coerceNumberProperty(p));
+  set pageSizeOptions(
+    value: PageSizeOptions | readonly (number | {value: number; label: string})[],
+  ) {
+    this._pageSizeOptions = (value || []).map(p =>
+      typeof p === 'object'
```

Comment:
This can technically throw if `p` is null.

Reply:
that's right, good catch 🤦